### PR TITLE
[feat] ROM LDEVID DICE changes for MLDSA support

### PR DESCRIPTION
**Pull request by mhatrevi**

### CODE REVIEW

**Review by bluegate010 — DISMISSED** [review]

File: rom/dev/src/flow/cold_reset/ldev_id.rs [hunk 1]
```diff
@@ -125,25 +130,33 @@ impl LocalDevIdLayer {
...
-            report_boot_status(LDevIdKeyPairDerivationComplete.into());
         } else {
             cfi_assert!(result.is_err());
         }
-        result
+        let ecc_keypair = result?;
+
+        // Derive the MLDSA Key Pair.
+        let mldsa_key_pair =
+            Crypto::mldsa_key_gen(env, cdi, b"ldevid_mldsa_key", mldsa_keypair_seed)?;
```
[the remark this answers is not in the document]
Comment (mhatrevi) [reply rom/dev/src/flow/cold_reset/ldev_id.rs:156]: When I coded this up, the plan was to remove CFI from ROM hence I did not add the checks for this new code. Currently, it's disabled. That said, the new guidance it to bring it back. I have added the checks to the mldsa_key_gen.

File: rom/dev/src/flow/cold_reset/ldev_id.rs [hunk 2]
```diff
@@ -65,31 +67,34 @@ impl LocalDevIdLayer {
...
-                key_pair_seed: KEY_ID_LDEVID_MLDSA_KEYPAIR_SEED,
-                pub_key: Default::default(),
-            },
-            mldsa_subj_sn: [0; 64],
+            mldsa_subj_key_pair: mldsa_key_pair,
+            mldsa_subj_sn,
+            mldsa_subj_key_id,
         };
 
         // Generate Local Device ID Certificate
```
[the remark this answers is not in the document]
Comment (mhatrevi) [reply rom/dev/src/flow/cold_reset/ldev_id.rs:100]: That's correct. That change will come next week. Added a TODO for the same.

File: rom/dev/src/flow/cold_reset/x509.rs [hunk 3]
```diff
@@ -34,19 +37,50 @@ impl X509 {
...
+            PubKey::Mldsa(pub_key) => {
+                // pub_key is in Caliptra Hw format (big-endian DWORDs). Convert it to little-endian DWORDs.
+                copy_and_swap_endianess(pub_key.0.as_bytes(), pub_key_bytes);
+                pub_key_bytes.len()
+            }
+        }
+    }
+
+    fn pub_key_digest(env: &mut RomEnv, pub_key: &PubKey) -> CaliptraResult<Array4x8> {
+        let mut pub_key_bytes: [u8; size_of::<Mldsa87PubKey>()] = [0; size_of::<Mldsa87PubKey>()];
```
[the remark this answers is not in the document]
Comment (mhatrevi) [reply rom/dev/src/flow/cold_reset/x509.rs:67]: Added.

**Review by bluegate010 — APPROVED** [review]

**Review by swenson — APPROVED** [review]
LGTM

**Pull request merged**